### DISCUSSION TITLE
LOGINFOS: is there any chance to get log.h modify to send Messages to Sonde Serial Pin 1,2,3?

### DISCUSSION THREAD

**Discussion by whallmann** [mikaelnousiainen/RS41ng#66] [General]
Sorry but i am not able to install the hole arm-debugging tools as recomanded. I do not understand - including follow missing links for download - to get the arm addional gdb.  so i am searching for easier alternativs - I looked into the log.h & log.c  There are only defines to push most of it to printf. My idea was to try the Pins on the RS41  for Serial Pin 1 GND, Pin 2 TX, Pin 3 RX to use with a ttl/usb converter. But it seems there is nothing going out to this pins. Also i dont find any hints on the baud rate.  
Am i on the right way or is there no chance to easy get the log-entries out to this port without the hole bunch of arm-toolchain. 
**I dont need to debug or step through the programm. I only need the log infos out.**

**Comment by mikaelnousiainen** [comment]
Thanks for noting that the link to download the ARM toolchain is broken. I will fix it in the README.

The correct link is now: https://developer.arm.com/downloads/-/gnu-rm

Look for the latest version 10.3-2021.10 `arm-none-eabi` and pick the one for your operating system.

The reason why downloading the ARM GNU toolchain is required is that most operating systems (incl. Linux distros) do not have the GNU GDB debugger for the `arm-none-eabi` hardware target.

The only thing you need to do is to unpack the downloaded package and simply run the `arm-none-eabi-gdb` in the `bin` directory.

Regarding getting log output from the serial port: it is possible, but I'm personally not sure it is worth the effort. Additionally, it prevents the use of I²C bus and external sensors, since the external serial port shares pins with the I²C bus.

**Reply by whallmann** [reply]
Heavy - very heavy but i really got it - never believed this will happen. After some step stones i now see debug messages on the openOCD console.  So that's a way i am able to see more things what happens during the interrupt calls ;-).
Thanks for help so far Mikael.

**Comment by whallmann** [comment]
Ok this is answered. closing it. Thanks
